axum-extra/src/routing/resource.rs

```diff
@@ -52,27 +54,17 @@ pub struct Resource<B = Body> {
 }
 
 impl<B: Send + 'static> Resource<B> {
-    fn index_create_path(&self) -> String {
-        format!("/{}", self.name)
-    }
-
-    fn show_update_destroy_path(&self) -> String {
-        format!("/{0}/:{0}_id", self.name)
-    }
-
-    fn route<T>(mut self, path: &str, svc: T) -> Self
-    where
-        T: Service<Request<B>, Response = Response<BoxBody>, Error = Infallible>
-            + Clone
-            + Send
-            + 'static,
-        T::Future: Send + 'static,
-    {
-        self.router = self.router.route(path, svc);
-        self
+    /// Create a `Resource` with the given name.
+    ///
+    /// All routes will be nested at `/{resource_name}`.
+    pub fn named(resource_name: &str) -> Self {
+        Self {
+            name: resource_name.to_string(),
```

Comment:
Nitpick: I hate `str_ref.to_string()` because it uses the name of an operation that should be done very rarely (stringification of something that is not a string) in a very boring way and so results in one paying less attention to `.to_string()` calls overall.

I would recommend using `.into()` or `.to_owned()` instead.

There's also a clippy lint for this; if you agree with my reasoning you could turn it on for all the axum crates: https://rust-lang.github.io/rust-clippy/master/index.html#str_to_string

Reply:
It's mostly a habit of mine. I don't feel strongly about it. Feel free to add the lint and change existing callsites.